AdobeStockAsset/Model/AddIsDownloadedToSearchResult.php

```diff
@@ -0,0 +1,80 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+declare(strict_types=1);
+
+namespace Magento\AdobeStockAsset\Model;
+
+use Magento\Framework\Api\AttributeValueFactory;
+use Magento\Framework\Api\Search\SearchResultInterface;
+use Magento\Framework\App\ResourceConnection;
+
+/**
+ * Class AddIsDownloadedToSearchResult
+ */
+class AddIsDownloadedToSearchResult
```

Comment:
I would be good to use generic name for this service like `AppendAttributes` as later it could be reused to add additional information like `is_licensed` etc

Reply:
@sivaschenko agree. I have made the changes, thank you